src/lib/PingTimer.ts

```diff
@@ -1,28 +1,36 @@
-import timers from './timers'
+import getTimer from './get-timer'
+import type { TimerVariant } from './shared'
 
 export default class PingTimer {
 	private keepalive: number
 
 	private timer: any
 
+	private variant: TimerVariant
+
 	private checkPing: () => void
 
-	constructor(keepalive: number, checkPing: () => void) {
+	constructor(
+		keepalive: number,
+		checkPing: () => void,
+		variant: TimerVariant,
+	) {
 		this.keepalive = keepalive * 1000
 		this.checkPing = checkPing
+		this.variant = variant
 		this.reschedule()
 	}
 
 	clear() {
 		if (this.timer) {
-			timers.clear(this.timer)
+			getTimer(this.variant).clear(this.timer)
```

Comment:
@olso-nordsec Instead of calling `getTimer` everytime here I would set a private property `timer` on constructor with the value of getTimer based on the variant passed (the varian may be removed from pingtimer props then)

Reply:
changed